scripts/builtin/shapExplainer.dml

```diff
@@ -0,0 +1,730 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# Computes shapley values for multiple instances in parallel using antithetic permutation sampling.
+# The resulting matrix phis holds the shapley values for each feature in the column given by the index of the feature in the sample.
+#
+# This method first creates two large matrices for masks and masked background data for all permutations and
+# then runs in paralell on all instances in x.
+# While the prepared matrices can become very large (2 * #features * #permuations * #n_samples * #features),
+# the preparation of a row for the model call breaks down to a single element-wise multiplication of this mask with the row and
+# an addition to the masked background data, since masks can be reused for each instance.
+#
+# INPUT:
+# ---------------------------------------------------------------------------------------
+# model_function  The function of the model to be evaluated as a String. This function has to take a matrix of samples
+#                 and return a vector of predictions.
+#                 It might be usefull to wrap the model into a function the takes and returns the desired shapes and
+#                 use this wrapper here.
+# model_args      Arguments in order for the model, if desired. This will be prepended by the created instances-matrix.
+# x_instances     Multiple instances as rows for which to compute the shapley values.
+# X_bg            The background dataset from which to pull the random samples to perform Monte Carlo integration.
+# n_permutations  The number of permutaions. Defaults to 10. Theoretical 1 should already be enough for models with up
+#                 to second order interaction effects.
+# n_samples       Number of samples from X_bg used for marginalization.
+# remove_non_var  EXPERIMENTAL: If set, for every instance the varaince of each feature is checked against this feature in the
+#                 background data. If it does not change, we do not run any model cals for it.
+# seed            A seed, in case the sampling has to be deterministic.
+# verbose         A boolean to enable logging of each step of the function.
+# ---------------------------------------------------------------------------------------
+#
+# OUTPUT:
+# -----------------------------------------------------------------------------
+# S              Matrix holding the shapley values along the cols, one row per instance.
+# expected       Double holding the average prediction of all instances.
+# -----------------------------------------------------------------------------
+s_shapExplainer = function(String model_function, list[unknown] model_args, Matrix[Double] x_instances,
```

Comment:
Please reformat the function declarations so that you use 4 spaces for every parameter that is introduced in a new line and 2 spaces for the return value and move the  opening bracket `{` to a new line. Please double check for all other functions in this file. 

```
example = function(String x, Double y, ...
    Matrix[Double] z)
  return Double r
{
}
```